transport/http/dial_test.go

```diff
@@ -26,3 +31,21 @@ func TestNewClient(t *testing.T) {
 		t.Fatalf("got %s, want: %s", got, want)
 	}
 }
+
+func TestNewClient_NewAuthUniverseDomain(t *testing.T) {
+	os.Setenv("GOOGLE_CLOUD_UNIVERSE_DOMAIN", "example.com")
```

Comment:
nit: use https://pkg.go.dev/testing#T.Setenv. then you don't need to do this cleanup game.

Reply:
Fixed, thanks!